lib/restclient.rb

```diff
@@ -90,8 +90,23 @@ def self.options(url, headers={}, &block)
     Request.execute(:method => :options, :url => url, :headers => headers, &block)
   end
 
-  class << self
-    attr_accessor :proxy
+  # A global proxy URL to use for all requests. This can be overridden on a
+  # per-request basis by passing `:proxy` to RestClient::Request.
+  def self.proxy
+    @proxy
+  end
+  def self.proxy=(value)
+    @proxy = value
+    @proxy_set = true
+  end
+
+  # Return whether RestClient.proxy was set explicitly. We use this to
+  # differentiate between no value being set and a value explicitly set to nil.
+  #
+  # @return [Boolean]
+  #
+  def self.proxy_set?
+    !!@proxy_set
```

Comment:
Could someone explain the use of `!!` here, please?  This appears to be intentional and I'm likely missing a significant 'rubyism' OR this is just ridiculous and evaluates to `@proxy_set` anyway?  I suppose, if `@proxy_set.nil?` is true, then I would be inclined to use `@proxy_set || false`.  So, is that how ruby interprets the `!!@proxy_set`?